src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -2995,7 +2997,42 @@ private StatementAst ConfigurationStatementRule(IEnumerable<AttributeAst> custom
                         {
                             // Load the default CIM keywords
                             Collection<Exception> CIMKeywordErrors = new Collection<Exception>();
-                            Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscClassCache.LoadDefaultCimKeywords(CIMKeywordErrors);
+                            if (ExperimentalFeature.IsEnabled(Dsc.Json.DscClassCache.DscExperimentalFeatureName))
+                            {
+                                // In addition to checking if experimental feature is enabled
+                                // also check if PSDesiredStateConfiguration is already loaded
+                                // if pre-v3 is already loaded then use old mof-based APIs
+                                // otherwise use json-based APIs
+
+                                p.AddCommand(new CmdletInfo("Get-Module", typeof(Microsoft.PowerShell.Commands.GetModuleCommand)));
+                                p.AddParameter("Name", "PSDesiredStateConfiguration");
+                                
+                                bool prev3IsLoaded = false;
+                                foreach (PSModuleInfo moduleInfo in p.Invoke<PSModuleInfo>())
+                                {
+                                    if (moduleInfo.Version.Major < 3)
+                                    {
+                                        prev3IsLoaded = true;
+                                    }
+                                }
+
+                                p.Commands.Clear();
+                                
+                                useJsonSchema = !prev3IsLoaded;
+
+                                if (useJsonSchema)
+                                {
+                                    Dsc.Json.DscClassCache.LoadDefaultCimKeywords(CIMKeywordErrors);
```

Comment:
Good point. Thinking about a proper namespace name, the best options I got are `.CrossPlatform` or `.v3Utilities` instead of `.Json` part to show the difference with V2 MOF-based APIs.
Renamed here and in DSC module to `.CrossPlatform` for now. Please let me know if this does not work.